runelite-client/src/main/java/net/runelite/client/plugins/bankhistory/BankValueHistoryTracker.java

```diff
@@ -0,0 +1,319 @@
+		}
+	}
+
+	/**
+	 * Gets the current bank value history from disk.
+	 *
+	 * @param username username associated with the bank value.
+	 * @return the retrieved bank value
+	 */
+	public BankValueHistoryContainer getBankValueHistory(String username)
+	{
+		File playerHistoryFile = getFileForUser(username);
+		if (playerHistoryFile == null)
+		{
+			return null;
+		}
+
+		try (FileReader reader = new FileReader(playerHistoryFile))
+		{
+			log.debug("Creating bank history cache file at {}", playerHistoryFile.getAbsolutePath());
+			BankValueHistoryContainer container = GSON.fromJson(reader, BankValueHistoryContainer.class);
+			if (container == null)
+			{
+				return new BankValueHistoryContainer();
+			}
+
+			return container;
+		}
+		catch (IOException e)
+		{
+			log.error("Error reading/writing to the cache files", e);
+		}
+
+		return null;
+	}
+
+	/**
+	 * Get the data file for a specific user
+	 *
+	 * @param username user to get the data for
+	 * @return data file for user
+	 */
+	public File getFileForUser(String username)
+	{
+		File file = new File(HISTORY_CACHE, username + EXTENTION);
+		try
+		{
+			file.createNewFile();
+			if (panel != null)
+			{
+				panel.hidePanels();
+			}
+		}
+		catch (IOException e)
+		{
+			e.printStackTrace();
+		}
+
+		return file;
+	}
+
+
+	/**
+	 * Get all accounts that have tracking data from the local file cache.
+	 *
+	 * @return available accounts with data
+	 */
+	public List<String> getAvailableUsers()
+	{
+		List<String> result = new ArrayList<>();
+		File[] accountFiles = HISTORY_CACHE.listFiles();
+		if (accountFiles == null)
+		{
+			return result;
+		}
+
+		for (File file : accountFiles)
+		{
+			result.add(file.getName().replace(EXTENTION, ""));
+		}
+
+		return result;
+	}
+
+	/**
+	 * Gets the last time something was added to the file cache.
+	 *
+	 * @param username username
+	 * @param tab      number of the tab
+	 * @return LocalDateTime last data entry
+	 */
+	public LocalDateTime getLastDataEntry(String username, int tab)
+	{
+		BankValueHistoryContainer container = getBankValueHistory(username);
+
+		if (container == null)
+		{
+			return null;
+		}
+
+		Set<LocalDateTime> times =
+			container
+				.getPricesMap()
+				.entrySet()
+				.parallelStream()
+				.filter(bv -> bv.getValue().getTab() == tab)
+				.map(Map.Entry::getKey)
+				.collect(Collectors.toSet());
+
+		if (times.isEmpty())
+		{
+			return null;
+		}
+
+		return Collections.max(times);
+	}
+
+	/**
+	 * Add a new dataset entry.
+	 *
+	 * @param force force a new entry, regardless of user configuration or last entry time.
+	 */
+	public void addEntry(boolean force, Consumer<String> callback)
+	{
+		clientThread.invokeLater(() ->
+		{
+			int currentBankTab = client.getVar(Varbits.CURRENT_BANK_TAB);
+			LocalDateTime lastEntry = getLastDataEntry(client.getUsername(), currentBankTab);
+			LocalDateTime nextUpdateTime = LocalDateTime.now().plusHours(plugin.getDefaultDatasetEntry());
```

Comment:
```suggestion
			LocalDateTime nextUpdateTime = lastEntry.plusHours(plugin.getDefaultDatasetEntry());
```

Seems to fix the issue with it not creating automatic datapoints after the initial datapoint.

Reply:
With this change you can remove the `lastEntry == null ||` bit from `if (force || lastEntry == null || LocalDateTime.now().isAfter(nextUpdateTime))` on line 263 as well.